astropy/coordinates/baseframe.py

```diff
@@ -589,7 +589,7 @@ def __init__(self, *args, **kwargs):
             if repr_kwargs:
                 if repr_kwargs.get('distance', True) is None:
                     del repr_kwargs['distance']
-                if (self.representation == SphericalRepresentation and
+                if (issubclass(self.representation, SphericalRepresentation) and
```

Comment:
When I was digging into this code the other day I wondered about that myself.  Assumed there was some reason it had to be strict, but I guess not.